CMakeLists.txt

```diff
@@ -1,338 +1,69 @@
-else()
-  set(CMAKE_INSTALL_RPATH # must be done before add_executable()
-    "../lib/${CMAKE_INSTALL_FULL_LIBDIR}")
-endif()
-
-if(DEFINED ENV{SECUTILS_USE_ICV})
-  add_executable(icvutil
-    util/icvutil.c
+if(NOT("$ENV{OPENSSL_LIB}" STREQUAL ""))
+  if(TARGET OpenSSL::SSL)
+    set_target_properties(OpenSSL::SSL PROPERTIES
+      IMPORTED_LOCATION $ENV{OPENSSL_LIB}/libssl.so
     )
-  target_link_libraries(icvutil
-    ${LIBSECUTILS_NAME}
-    ${OPENSSL_LIBRARIES}
-    )
-endif()
-
-configure_file(${INC_DIR}/secutils_static_config.h.in
-               ${INC_DIR}/secutils_static_config.h)
-
-# set_property(TARGET ${LIBSECUTILS_NAME} PROPERTY C_STANDARD 90)
-
-set_target_properties(${LIBSECUTILS_NAME} PROPERTIES
-  VERSION ${LIBSECUTILS_VERSION} SOVERSION ${LIBSECUTILS_VERSION_MAJOR}
-  # not used as directory structure would not be preserved on installation:
-  # PUBLIC_HEADER "${INC_PUBLIC_HDRS}"
-)
+  endif()
 
-if(NOT TARGET build)
-  add_custom_target(build COMMAND make all)
+  if(TARGET OpenSSL::Crypto)
+    set_target_properties(OpenSSL::Crypto PROPERTIES
+      IMPORTED_LOCATION $ENV{OPENSSL_LIB}/libcrypto.so
+    )
+  endif()
 endif()
 
+option(SIEMENS_LIBSECUTILS_USE_UTA "Use UTA API" OFF)
+option(SIEMENS_LIBSECUTILS_USE_ICV "Use configuration ICV" OFF)
+option(SIEMENS_LIBSECUTILS_NO_TLS "Do not use TLS" OFF)
 
-# installation and uninstall
+add_subdirectory(src)
 
-if(DEFINED ENV{ROOTFS})
-  set(CMAKE_INSTALL_PREFIX $ENV{ROOTFS} CACHE PATH "comment" FORCE)
-endif()
+# =============================
+# Installing
+# =============================
+include(GNUInstallDirs)
 
-# see also https://gitlab.kitware.com/cmake/cmake/-/issues/24929 and
-# https://stackoverflow.com/questions/46490448/how-i-can-add-a-changelog-in-deb-with-cmake
-add_custom_command(
-    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/debian/changelog"
-    COMMAND gzip -cn9 "${CMAKE_CURRENT_SOURCE_DIR}/debian/changelog"
-    > "${CMAKE_CURRENT_BINARY_DIR}/changelog.gz"
-    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/changelog.gz"
+install(FILES libsecutils.svg
+  DESTINATION ${CMAKE_INSTALL_DOCDIR}
 )
-if(NOT TARGET changelog)
-  add_custom_target(changelog ALL DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/changelog.gz")
-endif()
-
-install(TARGETS ${LIBSECUTILS_NAME}
-  LIBRARY
-  DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}"
-  COMPONENT lib
-  )
 
-find_program(DOXYGEN doxygen)
-if(NOT TARGET doc AND DOXYGEN)
-  add_custom_command(
-    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/secutils/basic.h"
-    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-    COMMAND doxygen Doxyfile # required packages: doxygen graphviz
-    OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/doc/html" # BINARY_DIR woud be better
-    )
-  add_custom_target(doc ALL DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/doc/html")
-endif()
-
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/doc/html/."
-  OPTIONAL # as doxygen may not be available
-  DESTINATION "${CMAKE_INSTALL_FULL_DOCDIR}-dev/html"
-  COMPONENT dev
-  )
```

Comment:
`make doc` does nothing - some definitions need to be re-added.

Reply:
`make -f Makefile_v1 doc` runs fine to me. There is no main `Makefile` or you mean something else?

Reply:
`make doc` should work as well, but it does nothing if the `doc` folder exists, and otherwise fails.

Reply:
I see. You mean `Makefile` generated by CMake?

Doxygen added into CMake. It should work now.

Reply:
Mostly. I had to add an `if(Doxygen_FOUND)`
and replace `set(DOXYGEN_OUTPUT_DIRECTORY doc/html)` by `set(DOXYGEN_OUTPUT_DIRECTORY doc)`